src/Domain/Model/AddressChange.php

```diff
@@ -112,12 +112,18 @@ public function isExported(): bool {
 	}
 
 	public function isModified(): bool {
-		return $this->createdAt < $this->modifiedAt;
+		return $this->createdAt < $this->modifiedAt || $this->hasBeenUsed();
```

Comment:
you could add another unit test, asserting that "isModified" should now consider this new condition (address change happened)

Reply:
I think I'll rather undo the change in this line, here is why:

Each method that changes the entity sets a new date. My thinking was that there *might* be a very fast test where `createdAt` and `modifiedAt` might have the same timestamp, making the test flaky, so I just slapped `hasBeenUsed` at the end because I know that for this case it would make a test less flaky. But I've never encountered a failing test on this, just making assumptions. After some research I now realize that PHP DateTime objects have microsecond precision, which means that even in a tight for loop the times will be different. Basically I've added code for a test case that will most likely never happen, also making this almost impossible to actually test (without unnecessary refactoring that passes a `Clock` instance to the class). In production, `createdAt` and `modifiedAt` will be *very* far apart anyway.